tests/benchmark/test_worst_bytecode.py

```diff
@@ -315,3 +318,188 @@ def test_worst_initcode_jumpdest_analysis(
+        + (Op.CODECOPY(size=Op.DUP1) if non_zero_data else Bytecode())
+        + Op.RETURN(0, Op.DUP1)
+    )
+    if non_zero_data:  # Pad to code_size.
+        code += bytes([i % 256 for i in range(code_size - len(code))])
+
+    initcode_template_contract = pre.deploy_contract(code=code)
+
+    # Create the benchmark contract which has the following design:
+    # ```
+    # PUSH(value)
+    # [EXTCODECOPY(full initcode_template_contract) -- Conditional that non_zero_data is True]`
+    # JUMPDEST (#)
+    # (CREATE|CREATE2)
+    # (CREATE|CREATE2)
+    # ...
+    # JUMP(#)
+    # ```
+    code_prefix = (
+        Op.EXTCODESIZE(address=initcode_template_contract)
+        + Op.PUSH1(value)
+        + Op.EXTCODECOPY(
+            address=initcode_template_contract,
+            size=Op.DUP2,  # DUP2 refers to the EXTCODESIZE value above.
+        )
+    )
+
+    if opcode == Op.CREATE2:
+        # For CREATE2, we provide an initial salt.
+        code_prefix = code_prefix + Op.PUSH1(42)
+
+    attack_block = (
+        # For CREATE:
+        # - DUP2 refers to the EXTOCODESIZE value  pushed in code_prefix.
+        # - DUP3 refers to PUSH1(value) above.
+        Op.POP(Op.CREATE(value=Op.DUP3, offset=0, size=Op.DUP2))
+        if opcode == Op.CREATE
+        # For CREATE2: we manually push the arguments because we leverage the return value of
+        # previous CREATE2 calls as salt for the next CREATE2 call.
+        #  - DUP4 is targeting the PUSH1(value) from the code_prefix.
+        #  - DUP3 is targeting the EXTCODESIZE value pushed in code_prefix.
+        else Op.DUP3 + Op.PUSH0 + Op.DUP4 + Op.CREATE2
+    )
+    code = code_loop_precompile_call(code_prefix, attack_block, fork)
+
+    tx = Transaction(
+        # Set enough balance in the pre-alloc for `value > 0` configurations.
+        to=pre.deploy_contract(code=code, balance=1_000_000_000 if value > 0 else 0),
+        gas_limit=env.gas_limit,
+        sender=pre.fund_eoa(),
+    )
+
+    state_test(
+        env=env,
+        pre=pre,
+        post={},
+        tx=tx,
+    )
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.CREATE,
+        Op.CREATE2,
+    ],
+)
+def test_worst_creates_collisions(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    opcode: Op,
+):
+    """Test the CREATE and CREATE2 collisions performance."""
+    env = Environment()
+
+    # We deploy a "proxy contract" which is the contract that will be called in a loop
+    # using all the gas in the block. This "proxy contract" is the one executing CREATE2
+    # failing with a collision.
+    # The reason why we need a "proxy contract" is that CREATE(2) failing with a collision will
+    # consume all the available gas. If we try to execute the CREATE(2) directly without being
+    # wrapped **and capped in gas** in a previous CALL, we would run out of gas very fast!
+    init_code = Op.PUSH0 + Op.PUSH0 + Op.RETURN
```

Comment:
This can also be just `Op.STOP` or even empty initcode (they will, just like this, leave an empty contract if creation is succesful (but will bump the nonce of target account to 1))